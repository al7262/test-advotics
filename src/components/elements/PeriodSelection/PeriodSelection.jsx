/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import moment from 'moment';
import calendar from '../../../assets/calendar.png';
import usePopper from '../../../hooks/usePopper';
import Button from '../Button';
import Popper from '../Popper';
import CalendarDaily from '../CalendarDaily';
import styles from './index.css';

export default function PeriodSelection() {
  const [startDate, setStartDate] = useState(moment().add(-7, 'days'));
  const [endDate, setEndDate] = useState(moment().add(-1, 'days'));
  const combinedDate = `${startDate.format('DD MMMM YYYY')} - ${endDate.format('DD MMMM YYYY')}`;
  const [date, setDate] = useState(combinedDate);
  const [open, setOpen] = useState(false);
  const [chosen, setChosen] = useState('Last 7 days');
  const disableAfterEndDate = moment(startDate).add(6, 'M').isAfter(moment()) ? moment().add(-1, 'd') : moment(startDate).add(6, 'M');

  const onClick = () => setOpen(!open);
  const onApply = () => {
    setOpen(false);
    setDate(combinedDate);
  };

  const updateDate = (choose, start, end) => {
    setStartDate(moment().add(-start, 'days'));
    setEndDate(moment().add(-end, 'days'));
    setChosen(choose);
  };

  const choices = [
    {
      label: 'Today',
      disabled: true,
      action: () => updateDate('Today'),
    },
    {
      label: 'Yesterday',
      action: () => updateDate('Yesterday', 1, 1),
    },
    {
      label: 'Last 7 days',
      action: () => updateDate('Last 7 days', 7, 1),
    },
    {
      label: 'Last 30 days',
      action: () => updateDate('Last 30 days', 30, 1),
    },
    {
      label: 'This Month',
      action: () => {
        setStartDate(moment().startOf('month'));
        setEndDate(moment());
        setChosen('This Month');
      },
    },
  ];

  const popperDate = usePopper();

  const rootStyles = [
    styles.root,
    open && styles.open,
  ].filter(Boolean).join(' ');

  return (
    <div className={rootStyles}>
      <div
        className={styles.summary}
        ref={popperDate.buttonRef}
        onClick={onClick}
      >
        <img src={calendar} alt="calendar" />
        <h5>Period</h5>
        <h5>
          {!open && date}
        </h5>
        <Button variant="link" onClick={onClick}>
          <span />
        </Button>
      </div>
      <Popper
        className={styles.popper}
        onClose={() => setOpen(false)}
        open={open}
        {...{ ...popperDate, style: {} }}
      >
        <section className={styles.choice}>
          {choices.map((data, key) => (
            <Button
              className={[styles['choice-item'], data.label === chosen ? styles.active : ''].join(' ')}
              variant="link"
              onClick={data.action}
              key={`choices-${key + 1}`}
              disabled={data.disabled}
            >
              <p>{data.label}</p>
            </Button>
          ))}
          <Button size="small" fixed onClick={onApply}>
            <p>Apply</p>
          </Button>
        </section>
        <CalendarDaily
          onSelect={(v) => { setStartDate(moment(v)); setChosen(''); }}
          date={startDate}
          disabledAfter={endDate}
          disabledBefore={moment(endDate).add(-6, 'M')}
        />
        <CalendarDaily
          onSelect={(v) => { setEndDate(moment(v)); setChosen(''); }}
          date={endDate}
          disabledAfter={disableAfterEndDate}
          disabledBefore={startDate}
        />
      </Popper>
    </div>
  );
}
