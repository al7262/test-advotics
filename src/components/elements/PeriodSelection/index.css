.root {
    height: 3rem;
    margin-left: auto;
    position: relative;
}

.summary {
    border: 0;
    height: 100%;
    width: 25rem;
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 1.5rem;
    padding: .75rem 1rem;
    background-color: var(--white-color);
    box-shadow: 0px 2px 3px #00000029;
    border-radius: 2px;
    transition: ease-in-out .5s;
}

.open > .summary {
    width: 37.5rem;
}

.summary > img {
    height: 100%;
}

.summary > h5 {
    color: var(--black-color-2);
    transition: ease-in-out .5s;
}

.summary > h5:first-of-type {
    color: var(--black-color-3);
}

.summary > h5:last-of-type {
    flex: 1;
}

.summary > button {
    all: unset;
    display: flex;
    padding-bottom: .125rem;
}

.summary > button > span {
    border: solid var(--black-color-2);
    border-width: 0 .125rem .125rem 0;
    display: inline-block;
    padding: .25rem;
    transform: rotate(45deg);
    transition: ease-in-out .5s;
}

.open > .summary > button > span {
    transform: rotate(-135deg);
}

.popper {
    top: 3rem;
    right: 0;
    left: 0;
    width: 100%;
    overflow: hidden;
    padding: 1rem;
    box-shadow: 0px 2px 3px #00000029;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: 1rem;
    align-items: flex-start;
}

.choice {
    padding-right: 1rem;
    border-right: .0625rem solid var(--black-color-4);
}

.choice > .choice-item {
    font-weight: 400;
    color: var(--black-color-3) !important;
    padding: .75rem 0;
    border-bottom: .0625rem solid var(--black-color-4);
    width: 100%;
    justify-content: left;
}

.choice > .choice-item:hover > p {
    color: var(--primary-color);
}

.choice > .choice-item:disabled > p {
    color: var(--black-color-5);
}

.choice > .choice-item.active > p {
    color: var(--primary-color-dark);
    font-weight: 600;
}

.choice > .choice-item:first-child {
    padding-top: 0;
}

.choice > .choice-item:last-child {
    border-bottom: 0;
}


