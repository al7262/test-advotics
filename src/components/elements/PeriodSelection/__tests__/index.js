/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PeriodSelection from '../PeriodSelection';

describe('src/components/elements/PeriodSelection', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PeriodSelection />);
    expect(tree).toMatchSnapshot();
  });
});
