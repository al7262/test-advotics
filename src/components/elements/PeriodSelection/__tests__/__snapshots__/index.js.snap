// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/PeriodSelection render 1`] = `
<div
  className=""
>
  <div
    onClick={[Function]}
  >
    <img
      alt="calendar"
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <h5>
      Period
    </h5>
    <h5>
      27 June 2021 - 03 July 2021
    </h5>
    <Button
      className=""
      color="primary"
      disabled={false}
      fixed={false}
      icon={null}
      isLoading={false}
      onClick={[Function]}
      rounded={false}
      size="medium"
      to=""
      type="button"
      variant="link"
    >
      <span />
    </Button>
  </div>
  <Popper
    buttonRef={
      Object {
        "current": Object {},
      }
    }
    className=""
    onClose={[Function]}
    open={false}
    popperRef={null}
    style={Object {}}
  >
    <section>
      <Button
        className=" "
        color="primary"
        disabled={true}
        fixed={false}
        icon={null}
        isLoading={false}
        onClick={[Function]}
        rounded={false}
        size="medium"
        to=""
        type="button"
        variant="link"
      >
        <p>
          Today
        </p>
      </Button>
      <Button
        className=" "
        color="primary"
        disabled={false}
        fixed={false}
        icon={null}
        isLoading={false}
        onClick={[Function]}
        rounded={false}
        size="medium"
        to=""
        type="button"
        variant="link"
      >
        <p>
          Yesterday
        </p>
      </Button>
      <Button
        className=" "
        color="primary"
        disabled={false}
        fixed={false}
        icon={null}
        isLoading={false}
        onClick={[Function]}
        rounded={false}
        size="medium"
        to=""
        type="button"
        variant="link"
      >
        <p>
          Last 7 days
        </p>
      </Button>
      <Button
        className=" "
        color="primary"
        disabled={false}
        fixed={false}
        icon={null}
        isLoading={false}
        onClick={[Function]}
        rounded={false}
        size="medium"
        to=""
        type="button"
        variant="link"
      >
        <p>
          Last 30 days
        </p>
      </Button>
      <Button
        className=" "
        color="primary"
        disabled={false}
        fixed={false}
        icon={null}
        isLoading={false}
        onClick={[Function]}
        rounded={false}
        size="medium"
        to=""
        type="button"
        variant="link"
      >
        <p>
          This Month
        </p>
      </Button>
      <Button
        className=""
        color="primary"
        disabled={false}
        fixed={true}
        icon={null}
        isLoading={false}
        onClick={[Function]}
        rounded={false}
        size="small"
        to=""
        type="button"
        variant="filled"
      >
        <p>
          Apply
        </p>
      </Button>
    </section>
    <CalendarDaily
      date={"2021-06-27T07:44:49.689Z"}
      disabledAfter={"2021-07-03T07:44:49.691Z"}
      disabledBefore={"2021-01-03T07:44:49.691Z"}
      name=""
      onSelect={[Function]}
    />
    <CalendarDaily
      date={"2021-07-03T07:44:49.691Z"}
      disabledAfter={"2021-07-03T07:44:49.693Z"}
      disabledBefore={"2021-06-27T07:44:49.689Z"}
      name=""
      onSelect={[Function]}
    />
  </Popper>
</div>
`;
